### ISSUE TITLE
CA2101 is shown when MarshalAs is not LPStr or LPWStr

### ISSUE THREAD

**Issue by SupinePandora43** [dotnet/roslyn-analyzers#5009]
# Example Input
```cs
[DllImport("Ultralight")]
public static extern void ulConfigSetResourcePath(IntPtr config, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(ULStringMarshaler))] string resource_path);
```
# Output
![image](https://user-images.githubusercontent.com/36124472/114137835-31e79680-9926-11eb-9ba1-96d44b4977c1.png)

# Steps To Reproduce

1. define extern method with `DllImport` Attribute and `string` argument
2. specify marshalling using `[MarshalAs(UnmanagedType.LPUTF8Str)]` or `CustomMarshaler`
3. get `CA2101`

**Comment by Naine** [comment]
I'm finding this is raised even if MarshalAs is LPStr.

```csharp
[DllImport("mylib", CharSet = CharSet.Ansi, ExactSpelling = true, ThrowOnUnmappableChar = true)]
public static extern void Foo([MarshalAs(UnmanagedType.LPStr)] string s); // CA2102
```

The only way to satisfy the analyzer seems to be to marshal the string as unicode/utf16, which is not possible if the native API doesn't support that.
